File: backend/src/main/java/com/backend/ecommerce/Order/business/requests/CreateOrderRequest.java
package com.backend.ecommerce.Order.business.requests;

import com.backend.ecommerce.UpdatedCart.entities.entity.CartItem;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class CreateOrderRequest {
    private String name;
    private String surName;
    private String email;
    private String phoneNumber;
    private String province;
    private String district;
    private String address;
    private String paymentMethod;
    private String discountCode;
    private List<CartItem> cartItemList;
}

File: frontend/src/utils/Order/CreateOrderRequest.js
export class CreateOrderRequest {
  constructor({
    name,
    surName,
    email,
    phoneNumber,
    province,
    district,
    address,
    paymentMethod,
    discountCode = "",
    cartItemList = [],
  }) {
    this.name = name;
    this.surName = surName;
    this.email = email;
    this.phoneNumber = phoneNumber;
    this.province = province;
    this.district = district;
    this.address = address;
    this.paymentMethod = paymentMethod;
    this.discountCode = discountCode;
    this.cartItemList = cartItemList;
  }

  toRequestObject() {
    return {
      name: this.name,
      surName: this.surName,
      email: this.email,
      phoneNumber: this.phoneNumber,
      province: this.province,
      district: this.district,
      address: this.address,
      paymentMethod: this.paymentMethod,
      discountCode: this.discountCode,
      cartItemList: this.cartItemList,
    };
  }
}
